### ISSUE TITLE
Enhancement: Create a tool for the creation of OpMon

### ISSUE THREAD

**Issue by PyroFlareX** [OpMonTeam/OpMon-Archive#95]
I was looking through the project, and there weren't any resources for existing OpMon besides a JSON that didn't really have the names. Are there any current resources or example sheets for the creation of OpMon? If not, a simple (even a commandline) tool that exports an OpMon JSON based on given info. I'd make one, but can't seem to find any info on what the format is.

If it hasn't been done already, having a JSON or a custom file format extension PER OpMon would make creation simpler as well.

EG: OpMon/Data/OpMon/*.json (or *.opmn) with a file for each OpMon.

**Comment by milkyroute** [comment]
There is a complete guide about creating OpMons here in the wiki: https://github.com/OpMonTeam/OpMon/wiki/How-to-create-an-OpMon
However, the JSON format is not explained very well so I will try to make a guide as soon as possible. OpMon is currently on hiatus but I should have at least the time to create a guide explaining how the JSON works in the week, probably Wednesday. Thanks for noticing this problem!
For the OpMon creator, it's a very good idea. If you're interested in doing it, that would be great! Else,  I currently prefer to concentrate on the 0.16 so I will do it after.
And about doing a file for each OpMon, I must admit I don't know if I should do it or not since it means making a big amount of files, and I don't really see what it would bring more than putting all of them in one file. I will think about it.

**Comment by PyroFlareX** [comment]
I started to build the tool here:
https://github.com/PyroFlareX/OpMon-Creation-Tool

**Comment by cyrielcentori** [comment]
Oh, thank you! Meanwhile, I've done a little guide about the OpMon JSON format. I will add other formats guides, but for now, this is enough I think.

**Comment by milkyroute** [comment]
Oh, thank you! Meanwhile, I've done a little guide about the OpMon JSON format. I will add other formats guides, but for now, this is enough I think : https://github.com/OpMonTeam/OpMon/wiki/JSON-guide

**Comment by Srykah** [comment]
This tool probably should be a new repository of the OpMonTeam account, don't you think ? A fork would be enough ^^

**Comment by Srykah** [comment]
Btw, I think it would be advised to do the user interface in Qt, as it will be more native for end-users and considerably speed up the coding process as well

**Comment by milkyroute** [comment]
I agree about Qt, I've already used it in the past for personal projects. However, I don't know anything about the Qt license, and if it's compatible with the GPL 3.0. And I've forked the repository, but I won't have to time to personally contribute to it.
@PyroFlareX : What do you think about it? Do you prefer being a collaborator in a repo in OpMonTeam or keeping you repo and OpMonTeam having a fork of it?

**Comment by Srykah** [comment]
> 
> 
> I don't know anything about the Qt license, and if it's compatible with the GPL 3.0

Qt has two licences : a proprietary one and a (L)GPL v3. So it is possible to do open-source projects based on Qt.

As for @PyroFlareX , maybe put him administrator of the fork, so he can manage it from the OpMon Team's group directly ?

**Comment by milkyroute** [comment]
That's what I've done, I invited him to the @OpMonTeam/opmon-creation-tool team, so he can have admin permissions on the fork. And ok for Qt!

**Comment by navet56** [comment]
Done
